be/src/exec/pipeline/exchange/mcast_local_exchange.cpp

```diff
@@ -0,0 +1,206 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/pipeline/exchange/mcast_local_exchange.h"
+
+#include "util/logging.h"
+
+namespace starrocks {
+namespace pipeline {
+
+MultiCastLocalExchanger::MultiCastLocalExchanger(size_t consumer_number)
+        : _mutex(),
+          _consumer_number(consumer_number),
+          _current_acc_chunk_size(0),
+          _progress(consumer_number),
+          _opened_source_opcount(consumer_number) {
+    Cell* dummy = new Cell();
+    _head = dummy;
+    _tail = dummy;
+    for (size_t i = 0; i < consumer_number; i++) {
+        _progress[i] = _tail;
+        _opened_source_opcount[i] = 0;
+    }
+}
+
+MultiCastLocalExchanger::~MultiCastLocalExchanger() {
+    while (_head != nullptr) {
+        auto* t = _head->next;
+        delete _head;
+        _head = t;
+    }
+}
+
+bool MultiCastLocalExchanger::can_push_chunk() const {
+    std::unique_lock l(_mutex);
+    if ((_current_acc_chunk_size - _fast_acc_chunk_size) > config::vector_chunk_size) {
+        return false;
+    }
+    return true;
+}
+
+Status MultiCastLocalExchanger::push_chunk(const vectorized::ChunkPtr& chunk, int32_t sink_driver_sequence) {
+    if (chunk->num_rows() == 0) return Status::OK();
+
+    std::unique_lock l(_mutex);
+
+    int32_t used_count = (_consumer_number - _opened_source_number);
+
+    auto* cell = new Cell();
+    cell->chunk = chunk;
+    cell->used_count = used_count;
+    cell->acc_chunk_size = _current_acc_chunk_size;
+    cell->next = nullptr;
+
+    _tail->next = cell;
+    _tail = cell;
+    _current_acc_chunk_size += chunk->num_rows();
```

Comment:
It's use for compute row size? It's different with his name.

Reply:
Yes, it means how many rows/tuples are there. I think the name is fine, because we also have configuration `config::vector_chunk_size`. There is no ambiguity.

Reply:
But you use `_current_acc_chunk_size ` to compute `acc_chunk_size` in line 51, then compute `_current_acc_chunk_size` from `chunk->num_rows(); `, it's confusing.

Reply:
And you are right, `config::vector_chunk_size` is mostly used for chunk initialization, maybe `config::vector_chunk_capacity` is better than `config::vector_chunk_size`